apps/qubit/modules/digitalobject/actions/showComponent.class.php

```diff
@@ -74,13 +74,8 @@ public function execute($request)
      */
     private function checkShowGenericIcon()
     {
-        $resourceObject = $this->resource->object;
-
         // Get the parent resouce (IO or Actor) for Acl check
-        // if resource->object is empty and parent exists
-        if (!$resourceObject && $this->resouce->parent) {
-            $resourceObject = $this->resource->parent->object;
-        }
+        $resourceObject = $this->resource->parent->object;
```

Comment:
Do you really need `$resourceObject` as a var? I'd do it directly inside the switch on each call. That way you prevent the possibility of somebody adding master usage type to the switch and using the parent when it shouldn't.